src/modules/commander/HealthAndArmingChecks/checks/estimatorCheck.cpp

```diff
@@ -765,6 +766,44 @@ void EstimatorChecks::setModeRequirementFlags(const Context &context, bool pre_f
 		!checkPosVelValidity(now, xy_valid, gpos.eph, _param_com_pos_fs_eph.get(), gpos.timestamp,
 				     _last_gpos_fail_time_us, !failsafe_flags.global_position_invalid);
 
+	// Additional warning if the system is about to enter position-loss failsafe after dead-reckoning period
+	const float eph_critical = 2.5f * _param_com_pos_fs_eph.get(); // threshold used to trigger the navigation failsafe
+	const float gpos_critical_warning_thrld = math::max(0.9f * eph_critical, math::max(eph_critical - 10.f, 0.f));
+
+	estimator_status_flags_s estimator_status_flags;
+
+	if (_estimator_status_flags_sub.copy(&estimator_status_flags)) {
```

Comment:
Probably worth checking that it's a relatively recent esitmator_status_flags (eg timestamp within a few seconds).

Reply:
Okay done, with a 5s check. 
I'm curious though, why do you think that it's particularly here necessary?

Reply:
> I'm curious though, why do you think that it's particularly here necessary?

For all of these health checks there's the possibility that the estimator (or whatever) actually dies and at a minimum we don't want to get stuck reporting things on stale data (the last publication).